controllers/backplaneconfig_controller.go

```diff
@@ -1612,18 +1629,47 @@ func (r *MultiClusterEngineReconciler) finalizeBackplaneConfig(ctx context.Conte
 
 		return ctrl.Result{}, fmt.Errorf(
 			"waiting for 'open-cluster-management-hub' namespace to be terminated before proceeding with uninstallation")
-	} else if err != nil && !apierrors.IsNotFound(err) { // Return error, if error is not not found error
+
+	} else if !apierrors.IsNotFound(err) { // Return error, if error is not not found error
 		return ctrl.Result{}, err
 	}
 
 	globalSetNamespace := &corev1.Namespace{}
-	err = r.Client.Get(ctx, types.NamespacedName{Name: "open-cluster-management-global-set"}, globalSetNamespace)
-	if err == nil {
-		err := r.Client.Delete(ctx, globalSetNamespace)
-		if err != nil {
+	if err = r.Client.Get(ctx, types.NamespacedName{Name: "open-cluster-management-global-set"}, globalSetNamespace); err == nil {
```

Comment:
All these `if err == nil` bits would be a lot more readable if they were flipped to early-return `if err != nil` instead